src/main/java/bookface/logic/parser/LoanCommandParser.java

```diff
@@ -33,17 +38,48 @@ public LoanCommand parse(String args) throws ParseException {
 
         String[] nameKeywords = trimmedArgs.split("\\s+");
 
-
         try {
             String firstIndex = nameKeywords[0];
             String secondIndex = nameKeywords[1];
             Index userIndex = ParserUtil.parseIndex(firstIndex);
             Index bookIndex = ParserUtil.parseIndex(secondIndex);
-            return new LoanCommand(userIndex, bookIndex);
+
+            if (nameKeywords.length < 3) {
+                return new LoanCommand(userIndex, bookIndex);
+            }
         } catch (ParseException pe) {
             throw new ParseException(
                     String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT, LoanCommand.MESSAGE_USAGE), pe);
         }
+
+        try {
+            String firstIndex = nameKeywords[0];
+            String secondIndex = nameKeywords[1];
+            Index userIndex = ParserUtil.parseIndex(firstIndex);
+            Index bookIndex = ParserUtil.parseIndex(secondIndex);
+            if (nameKeywords.length == 3) {
+                String returnDate = nameKeywords[2];
+                List<Date> parsedReturnDate = new PrettyTimeParser().parse(returnDate);
+                if (parsedReturnDate.isEmpty()) {
```

Comment:
Since lines 63 to 66 are used again, we can do them after the if-else block

Reply:
Resolved.